datafusion/src/scalar.rs

```diff
@@ -37,58 +37,60 @@ use std::{convert::TryFrom, fmt, iter::repeat, sync::Arc};
 /// This is the single-valued counter-part of arrow’s `Array`.
 #[derive(Clone)]
 pub enum ScalarValue {
+    /// null value, typed or untyped
+    Null(Box<DataType>),
```

Comment:
What is the reason to use `Box<DataType>` here? Is to keep the size of `ScalarValue` relatively small?

What would you think about using `Arc<DataType>` instead to avoid as much copying?